### PR TITLE
[STORM-3675] Check worker process death using /proc/<pid> directory

### PR DESCRIPTION
**Pull request by bipinprasad**
## What is the purpose of the change

*This function can replace a system "ps" command with existence/ownership check on /proc/ directory. In additional, on loaded nodes, ps command can take a few seconds. java.io.File.exists() is expected to be much faster. Note that File watcher does not work on /proc file system and the event based approach was coded/tested and then discarded.*

## How was the change tested

*new test methods in ServerUtilsTest. Also run storm, kill worker and observe log to ensure proper detection*

### CODE REVIEW

**Comment by Ethanlm** [comment]
Unit test is failing 
```
classname: org.apache.storm.utils.ServerUtilsTest / testname: testIsAnyProcessPosixProcessPidDirAlive
java.lang.AssertionError: There are 5 failures in test
```

**Comment by bipinprasad** [comment]
Fixed the test for JDK11.

File: storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java [hunk 1]
```diff
@@ -162,6 +174,163 @@ public void testGetUserId() throws Exception {
...
+        int maxPidCnt = 5;
+        if (ServerUtils.IS_ON_WINDOWS) {
+            LOG.info("{}: test cannot be run on Windows. Marked as successful", testName);
+            return;
+        }
+        final Path parentDir = Paths.get("/proc");
+        if (!parentDir.toFile().exists()) {
+            LOG.info("{}: test cannot be run on system without process directory {}, os.name={}",
+                    testName, parentDir, System.getProperty("os.name"));
+            {
```
Comment (Ethanlm) [inline storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java:196]: I would remove "{". It doesn't seem necessary

File: storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java [hunk 2]
```diff
@@ -162,6 +174,163 @@ public void testGetUserId() throws Exception {
...
+                // check if we can get process id on this Posix system - testing test code, useful on Mac
```
Comment (Ethanlm) [inline storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java:197]: Can you please explain why is it "useful on Mac"?

File: storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java [hunk 3]
```diff
@@ -162,6 +174,163 @@ public void testGetUserId() throws Exception {
...
+                LOG.error(e);
+                continue;
+            }
+            observables.add(pid);
+        }
+        String userName = System.getProperty("user.name");
+        // now kill processes one by one
+        List<Long> pidList = new ArrayList<>(observables);
+        final long processKillIntervalMs = 2000;
+        for (int i = 0 ; i < pidList.size() ; i++) {
```
Comment (Ethanlm) [inline storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java:226]: nit: space before ";" can be removed

File: storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java (hunk 2 above)
Comment (bipinprasad) [reply storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java:197]: Rest of test will not get exercised on Mac (no /proc dir), but at least the getPidOfPosixProcess() will be validated locally.

File: storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java (hunk 3 above)
Comment (bipinprasad) [reply storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java:226]: fixed

File: storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java (hunk 1 above)
Comment (bipinprasad) [reply storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java:196]: fixed

**Pull request merged**